### PR TITLE
Us038 add a linting check

### PR DESCRIPTION
**Pull request by kelvinkainos**

### CODE REVIEW

File: app.ts [hunk 1]
```diff
@@ -1,33 +1,32 @@
-const express = require('express');
-const path = require('path');
-const nunjucks = require('nunjucks');
```
Comment (glenn-hamilton) [inline app.ts:3]: I'm surprised the linter has asked for the `;` to be removed as normally it asks for them to be included

File: public/styles.css [hunk 2]
```diff
@@ -0,0 +1,100 @@
+@font-face {
```
Comment (glenn-hamilton) [inline public/styles.css:1]: Should this be included, as it seems it's coming from another MR?

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+module.exports = {
```
Comment (glenn-hamilton) [inline .eslintrc.js:1]: Can you double check these settings, as in `tsconfig` it has a target of `ES5` but the `env` here is allowing `es2021`

File: app.ts (hunk 1 above)
Comment (kelvinkainos) [reply app.ts:3]: Was initially using eslitn for formatting now changed to using prettier after looking at documentation
https://typescript-eslint.io/rules/semi/

Now using rule to add semicolons before lines that may cause ASI failures
https://prettier.io/docs/en/options.html#semicolons

File: public/styles.css (hunk 2 above)
Comment (kelvinkainos) [reply public/styles.css:1]: Branched off from branding branch to test stylelint for css files

File: .eslintrc.js (hunk 3 above)
Comment (kelvinkainos) [reply .eslintrc.js:1]: Addressed by adding ES5 plugin

**Comment by kelvinkainos** [comment]
Should I add husky to add pre commit hooks to run the linters and formatter?

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -4,28 +4,29 @@
 name: Node.js CI
 
 on:
-  push:
```
Comment (glenn-hamilton) [inline .github/workflows/node.js.yml]: Is this because of the linter? If so it can probably ignore config files

File: app.ts (hunk 1 above)
Comment (glenn-hamilton) [reply app.ts:3]: `semicolons before lines`?

File: package.json [hunk 5]
```diff
@@ -1,33 +1,48 @@
...
-  },
-  "homepage": "https://github.com/CAP-M-McD/Team1_Frontend.git#readme"
+    "name": "team1-frontend",
+    "version": "1.0.0",
+    "description": "",
+    "main": "app.js",
+    "scripts": {
+        "start": "npx tsc && cp -R views dist && cp -R public dist && node dist/app.js",
+        "test": "echo \"Error: no test specified\" && exit 1",
+        "lint": "htmlhint *.html **/*.html && stylelint *.css **/*.css && eslint *.ts **/*.ts && prettier . --check",
```
Comment (glenn-hamilton) [inline package.json:9]: Could you add a section to the `README.md` file to outline how to run the lint check and how to fix the linter issues.

Could you also add the lint check to run before running a commit as a git hook?

File: app.ts (hunk 1 above)
Comment (kelvinkainos) [reply app.ts:3]: following the JS standard codestyle https://standardjs.com/rules.html#semicolons

File: .github/workflows/node.js.yml (hunk 4 above)
Comment (kelvinkainos) [reply .github/workflows/node.js.yml]: reverted it and added to ignore

File: package.json (hunk 5 above)
Comment (kelvinkainos) [reply package.json:9]: updated readme and added pre-commit hook

**Review by chloe22simpson — APPROVED** [review]

**Pull request merged**